packages/gutenpride-template/src/src/edit.js.mustache

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress components that create the necessary UI elements for the block
+ *
+ * @see https://github.com/WordPress/gutenberg/tree/master/packages/components
+ */
+import { Placeholder, TextControl } from '@wordpress/components';
+
+/**
+ * Retrieves the translation of text.
+ *
+ * @see https://developer.wordpress.org/block-editor/packages/packages-i18n/
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#edit
+ *
+ * @param {Object} [props]           Properties passed from the editor.
+ * @param {string} [props.className] Class name generated for the block.
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( { attributes, className, setAttributes } ) {
+	return (
+		<TextControl
+			className={ className }
```

Comment:
I see it's similar to what we have in the `save` implementation. We should use `useBlockProps` here:
```suggestion
			useBlockProps()
```

It will require more changes in this file. Do you need help?

Reply:
Thanks @gziolo I've been struggling with this one. I tried the following code and tested it with a new template published to npm named `jsa_gutenpride_template_useprops_two` but after building the plugin and testing it in my local install, the classes are not getting applied to the div around `<TextControl>`. i would really appreciate your feedback! I'll push the code for this file anyhow for you to see/comment.
```js
/**
 * WordPress components that create the necessary UI elements for the block
 *
 * @see https://developer.wordpress.org/block-editor/packages/packages-components/
 */
import { Placeholder, TextControl } from '@wordpress/components';

/**
 * Retrieves the translation of text.
 *
 * @see https://developer.wordpress.org/block-editor/packages/packages-i18n/
 */
import { __ } from '@wordpress/i18n';

/**
 * React hook that is used to mark the block wrapper element.
 * It provides all the necessary props like the class name.
 *
 * @see https://developer.wordpress.org/block-editor/packages/packages-block-editor/#useBlockProps
 */
import { useBlockProps } from '@wordpress/block-editor';

/**
 * The edit function describes the structure of your block in the context of the
 * editor. This represents what the editor will render when the block is used.
 *
 * @see https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#edit
 *
 * @return {WPElement} Element to render.
 */
export default function Edit( { attributes, setAttributes } ) {
	const blockProps = useBlockProps();
	return (
	<div { ...blockProps }>
		<TextControl
			value={ attributes.message }
			onChange={ ( val ) => setAttributes( { message: val } ) }
		/>
	</div>
	);
}

```

Reply:
I will test it myself, now that I added local package templates, hurray! 😄

Reply:
I know what has happened. The changes applied to `edit` are correct.

We miss `apiVersion` in the `index.js` file.

https://github.com/WordPress/gutenberg/blob/873af04d46ede2268cca0ff46ca7101aa3e51fcc/packages/create-block/lib/templates/esnext/src/index.js.mustache#L36-L39

Reply:
Thanks!!